File: Data Structures & Algorithms/max-water-container/submission-2.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0;
        int r = heights.size() - 1;
        int maxV = 0;
        int minH, area;
        while (l < r) {
            minH = min(heights[l], heights[r]);
            area = minH * (r - l);
            maxV = max(area, maxV);
            cout << maxV << endl;
            if (heights[l] > heights[r]) {
                r--;
            } else {
                l++;
            }
        }
        return maxV;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0
        r = 1
        maxV = 0
        while r < len(prices):
            maxV = max(maxV, prices[r] - prices[l])
            if prices[r] < prices[l]:
                l = r
            r += 1
        return maxV